From: Rob Swindell <swindell@broadcom.com>
Date: Sun, 27 Dec 2015 18:19:20 -0500
Subject: [PATCH 03/74] bnxt_en: Add support for upgrading APE/NC-SI firmware
 via Ethtool FLASHDEV
Patch-mainline: v4.5-rc1
Git-commit: 2731d70fa9cbb62e45743171bf979784fb36778c
References: bsc#963575 FATE#320144

NC-SI firmware of type apeFW (10) is now supported.

Signed-off-by: Rob Swindell <swindell@broadcom.com>
Signed-off-by: Michael Chan <mchan@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/broadcom/bnxt/bnxt_ethtool.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/ethernet/broadcom/bnxt/bnxt_ethtool.c
+++ b/drivers/net/ethernet/broadcom/bnxt/bnxt_ethtool.c
@@ -818,6 +818,9 @@ static int bnxt_flash_firmware(struct ne
 	case BNX_DIR_TYPE_BOOTCODE_2:
 		code_type = CODE_BOOT;
 		break;
+	case BNX_DIR_TYPE_APE_FW:
+		code_type = CODE_MCTP_PASSTHRU;
+		break;
 	default:
 		netdev_err(dev, "Unsupported directory entry type: %u\n",
 			   dir_type);
